plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/categorizer/package-info.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/**
+ * Query Insights search query categorizor to collect metrics related to search queries
+ */
+package org.opensearch.plugin.insights.core.service.categorizer;
```

Comment:
Most of the QueryCategorizer logic is independent of QueryInsightsService. Hence, categorizer should not be contained within service IMO.